index.js

```diff
@@ -18,30 +19,48 @@ var startTime = function() {
   });
 };
 
+var potentiallyMigrateDatabase = function() {
+  return new Promise(function (resolve, reject) {
+    postgrator.setConfig({
+      migrationDirectory: __dirname + '/migrations',
+      driver: 'pg',
+      connectionString: process.env.POSTGRESQL_URL
+    });
+
+    postgrator.migrate('201505271423', function(err, migrations) {
+      if (err) {
+        reject(err);
+      } else {
+        postgrator.endConnection(function() {
+          resolve(migrations);
+        });
+      }
+    });
+  });
+};
+
 var loop = function () {
   var starttime;
   startTime()
     .then(function (time) {
       starttime = time;
     })
-
+    .then(potentiallyMigrateDatabase)
+    .then(function(migrations) {
+      console.log('Sucessfully migrated', migrations);
+    })
     .then(couch2pg)
     .then(function () {
       console.log('Imported successfully at ' + Date());
-    }, function (err) {
-      console.log('Import errored at ' + Date());
-      console.log(err);
     })
-
     .then(xmlforms)
     .then(function () {
       console.log('XML forms completed at ' + Date());
-    }, function (err) {
-      console.log('XML forms errored at ' + Date());
-      console.log(err);
     })
-
-    .then(function () {
+    .catch(function(err) {
+      console.error(err);
```

Comment:
Originally, errors thrown by couch2pg did not prevent xmlforms from running. It is true that xmlforms consumes couch2pg, but in practice, I have seen occasions where couch2pg produces more data than xmlforms can consume in an interval. In cases like that, it made sense to continue running xmlforms parsing even when couch2pg failed. Perhaps this no longer holds with the changes to couch2pg.

We should definitely throw a timestamp on here regardless.